### PR TITLE
Add 'test connection' button to integrations view

### PR DESCRIPTION
**Pull request by jpople**
Closes PROD-2129

### Description Of Changes

Add a "test connection" button to the new integrations UI on the list view and also detail view.

Screenshot 2024-06-06 at 16 19 47
Screenshot 2024-06-06 at 16 20 00
Screenshot 2024-06-06 at 16 20 18
Screenshot 2024-06-06 at 16 21 32

### Code Changes

* [ ] Add "test connection" buttons to IntegrationBox and integration detail view pages
* [ ] Add new `useTestConnection` hook to control both and provide data and toasts

### Steps to Confirm

* Navigate to `/integrations`
* Create a new BigQuery integration and provide it with credentials
* On integration detail view, click "test connection"; should see success toast and status message should update with current timestamp
* Return to list view and click "test connection"; same thing should happen
* Return to detail view and click "manage"
* Input any incorrect credentials JSON
* Click "test connection" again
* Should show failure toast and change to failure message

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [ ] Issue Requirements are Met
* [ ] Update `CHANGELOG.md`

### CODE REVIEW

File: clients/admin-ui/cypress/e2e/integration-management.cy.ts [hunk 1]
```diff
@@ -49,6 +55,13 @@ describe("Integration management for data detection & discovery", () => {
         cy.getByTestId("empty-state").should("not.exist");
       });
 
+      it("should be able to test connections by clicking the button", () => {
+        cy.getByTestId("integration-info-bq_integration").within(() => {
+          cy.getByTestId("test-connection-btn").click();
+          cy.wait("@testConnection");
```
Comment (gilluminate) [inline clients/admin-ui/cypress/e2e/integration-management.cy.ts:61]: do you think it's worth checking for the toast here as well?

**Review by gilluminate — APPROVED** [review]
Everything here looks really good. Just one minor suggestion on the Cypress test.

File: clients/admin-ui/cypress/e2e/integration-management.cy.ts (hunk 1 above)
Comment (jpople) [reply clients/admin-ui/cypress/e2e/integration-management.cy.ts:61]: @gilluminate I tried to do that originally but wasn't able to actually test for it-- there's an onQueryStarted helper method that runs when the test query is run (see `datastore-connection.slice.ts`) and it throws an uncaught error in the Cypress environment.  I wasn't able to figure out how to suppress or stub a response or anything to run more Cypress commands after the test query.

Comment (gilluminate) [reply clients/admin-ui/cypress/e2e/integration-management.cy.ts:61]: I see. That's unfortunate! Ship it!

**Pull request merged**